cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java

```diff
@@ -85,7 +89,7 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
       }
 
       try {
-        taskRunner.run(Id.Program.from(namespaceId, applicationId, programType, programId),
+        taskRunner.run(new ApplicationId(namespaceId, applicationId, appVersion).program(programType, programId),
```

Comment:
`ProgramId(ApplicationId appId, ProgramType type, String program)` is not visible outside of its package and I have to use this method to include appVersion